sdk/azcore/runtime/policy_bearer_token.go

```diff
@@ -72,8 +72,8 @@ func (b *BearerTokenPolicy) authenticateAndAuthorize(req *policy.Request) func(p
 
 // Do authorizes a request with a bearer token
 func (b *BearerTokenPolicy) Do(req *policy.Request) (*http.Response, error) {
-	if strings.ToLower(req.Raw().URL.Scheme) != "https" {
-		return nil, shared.NonRetriableError(errors.New("bearer token authentication is not permitted for non TLS protected (https) endpoints"))
+	if err := checkHTTPSForAuth(req); err != nil {
+		return nil, err
```

Comment:
Could we have a single policy to check for SSL rather than requiring each token policy to add this check?

Reply:
The requirement for the check is tightly coupled to authentication.  E.g. `azcore.NewClient()` doesn't add any kind of authentication policy to the pipeline, the caller must add it. I do see the value though in having the check be centralized. I will think on this a bit but, at present, I don't see an easy way to solve this.

Reply:
I suppose we could expose a new pipeline policy that performs the `https` check. It would have to be publicly exposed so that SDKs can take advantage of it. This approach would mean that _all_ SDKs that include an authentication policy in their pipelines would need to include this new policy. Put another way, the current approach lets us centralize the checks. A new pipeline policy would require updating all SDKs as required.

Reply:
Having a separate policy for `https` validation also means that the authentication policies are insecure by default.